cranelift-wasm/src/sections_translator.rs

```diff
@@ -351,3 +351,46 @@ pub fn parse_data_section<'data>(
 
     Ok(())
 }
+
+/// Parses the Name section of the wasm module.
+pub fn parse_name_section<'data>(
+    mut names: NameSectionReader<'data>,
+    environ: &mut dyn ModuleEnvironment<'data>,
+) -> WasmResult<()> {
+    while let Ok(subsection) = names.read() {
+        match subsection {
+            wasmparser::Name::Function(function_subsection) => {
+                if let Some(function_names) = function_subsection
+                    .get_map()
+                    .ok()
+                    .and_then(parse_function_name_subsection)
+                {
+                    for (index, name) in function_names {
```

Comment:
Since each subsection may appear at most once per spec, should we have a boolean that checks if this is the first time we've seen such a subsection, and ignore the following function name subsections if we've seen it before?

Reply:
Since other subsections are not being parsed at the moment, I decided to just use an early return here, since that would have the same effect without the need for an extra variable. If/when other subsections _are_ parsed in the future, figured that would be an easy enough change to make, and this reads cleanly for the time being.

Reply:
Sorry, I wasn't talking about other kinds of subsections, but if there were several names subsections in the same name section. That being said, we're not validating properly the rest of the sections anyway, so it could be something we'd improve in the future; let's not block this PR on this.